torchgeo/datasets/cloud_cover.py

```diff
@@ -0,0 +1,410 @@
+            download: if True, download dataset and store it in the root directory
+            api_key: a RadiantEarth MLHub API key to use for downloading the dataset
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        self.root = root
+        self.chip_size = chip_size
+        self.split = split
+        self.bands = bands
+        self.transforms = transforms
+        self.checksum = checksum
+
+        if download:
+            self._download(api_key)
+
+        if not self._check_integrity():
+            raise RuntimeError(
+                "Dataset not found or corrupted. "
+                + "You can use download=True to download it"
+            )
+
+        self.chip_paths = self._load_collections()
+
+    def __len__(self) -> int:
+        """Return the number of items in the dataset.
+
+        Returns:
+            length of dataset in integer
+        """
+        return len(self.chip_paths)
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Returns a sample from dataset.
+
+        Args:
+            Index: index to return
+
+        Returns:
+            data and label at given index
+        """
+        image = self._load_image(index)
+        label = self._load_target(index)
+        sample: Dict[str, Tensor] = {"image": image, "mask": label}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def _load_image(self, index: int) -> Tensor:
+        """Load all source images for a chip.
+
+        Args:
+            index: position of the indexed chip
+
+        Returns:
+            a tensor of stacked source image data
+        """
+        source_asset_paths = self.chip_paths[index]["source"]
+        images = []
+        for path in source_asset_paths:
+            with rasterio.open(path) as image_data:
+                image_array = image_data.read(1).astype(np.int32)
+                images.append(image_array)
+        image_stack: "np.typing.NDArray[np.int_]" = np.stack(images, axis=0)
+        image_tensor = torch.from_numpy(image_stack)
+        return image_tensor
+
+    def _load_target(self, index: int) -> Tensor:
+        """Load label image for a chip.
+
+        Args:
+            index: position of the indexed chip
+
+        Returns:
+            a tensor of the label image data
+        """
+        label_asset_path = self.chip_paths[index]["target"][0]
+        with rasterio.open(label_asset_path) as target_data:
+            target_img = target_data.read(1).astype(np.int32)
+
+        target_array: "np.typing.NDArray[np.int_]" = np.array(target_img)
+        target_tensor = torch.from_numpy(target_array)
+        return target_tensor
+
+    @staticmethod
+    def _validate_stac_object(object_path: str) -> bool:
```

Comment:
That was the intended purposes of the method. If the STAC object doesn't exist, would it be better to let Python native library's error handling raise these instead of the `RuntimeError` raised? In other words, do you believe this method should simply be removed?